src/main/java/com/w6/email/GmailClient.java

```diff
@@ -0,0 +1,114 @@
+package com.w6.email;
+
+import com.sun.mail.pop3.POP3SSLStore;
+
+import java.util.*;
+import javax.mail.Flags;
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.Store;
+import javax.mail.URLName;
+import javax.mail.search.FlagTerm;
+
+public class GmailClient {
+    private Store store;
+    private String username, password;
+    private Folder folder;
+
+    public GmailClient() {
+    }
+
+    public void setUserPass(String username, String password) {
+        this.username = username;
+        this.password = password;
+    }
+
+    public void connect() throws Exception {
```

Comment:
It should not be public. Class should have simple interfaces, all rest should be private